File: Moviq/Js/models/cartItem.js
/*jslint nomen: true*/
/*global define*/
define('models/cartItem', {
    init: function (ko) {
        "use strict";

        if (!ko) {
            throw new Error('Argument Exception: ko is required to init the cartItem module');
        }

        var CartItem = function (cartItem) {
            var $this = this;

            $this.setCartItemData = function (cartItem, cartItemData) {
                if (!cartItem) {
                    throw new Error('cannot extend the properties of undefined');
                }

                cartItemData = cartItemData || {};

                var type = cartItemData._type || 'cartItem';

                cartItem.guid = ko.observable(cartItemData.guid);
                cartItem.userGuid = ko.observable(cartItemData.userGuid);
                cartItem.productUid = ko.observable(cartItemData.productUid);
                cartItem.title = ko.observable(cartItemData.title || undefined);
                cartItem.price = ko.observable(cartItemData.price || undefined);
                cartItem.thumbnailLink = ko.observable(cartItemData.thumbnailLink || '/images/cartItems/default.png');
                cartItem.lastModified = ko.observable(cartItemData.lastModified);
                cartItem.purchaseDate = ko.observable(cartItemData.purchaseDate);
                cartItem.quantity = ko.observable(cartItemData.quantity);
            };

            if (cartItem) {
                $this.setCartItemData($this, cartItem);
            }
        };

        return CartItem;
    }
});

File: Moviq.Domain.CartItem/CartItem.cs
using Moviq.Interfaces.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Moviq.Domain.CartItems
{
    public class CartItem : ICartItem, IHelpCategorizeNoSqlData
    {
        public CartItem() 
        {
            this._type = "cart_item";
        }

        public Guid Guid { get; set; }
        public Guid UserGuid { get; set; }
        public string ProductUid { get; set; }
        public string Title { get; set; }
        public decimal Price { get; set; }
        public string ThumbnailLink { get; set; }
        public DateTime LastModified { get; set; }
        public DateTime PurchaseDate { get; set; }
        public int Quantity { get; set; }
        public string _type { get; set; }
    }
}
